3MI0800270/2023-03-07/zad4.cpp

```diff
@@ -0,0 +1,157 @@
+#include <fstream>
+#include <iostream>
+#include <cstring>
+using namespace std;
+
+/* union IPv4
+{
+    u_int32_t address;
+    u_int8_t arg[4];
+};
+
+union Time {
+    u_int32_t time;
+    struct {
+        u_int8_t hours;
+        u_int8_t minutes;
+        u_int8_t seconds;
+        u_int8_t millis;
+    }S;
+    //i saw online that this is undefined behavior, but i wanted to try it because it seems interesting
+};
+
+Time createTime(u_int8_t hours, u_int8_t minutes, u_int8_t seconds, u_int8_t millis) {
+    return Time{ .S = {hours,minutes,seconds,millis} };
+}
+Time createTime(u_int32_t time) {
+    return Time{ .time = time };
+}
+
+IPv4 createIp(u_int8_t ip1, u_int8_t ip2, u_int8_t ip3, u_int8_t ip4) {
+    return IPv4{ .arg = {ip1,ip2,ip3,ip4} };
+}
+IPv4 createIp(u_int32_t ip) {
+    return IPv4{ .address = ip };
+} */
+
+struct IPv4 {
+    int addr[4];
+};
+
+struct Time {
+    int nums[4];
+};
+
+struct Log {
+    char name[16] = "";
+    IPv4 ip;
+    Time time;
+};
+
+IPv4 readIP(char* ip) {
+    IPv4 res;
+    for (int i = 0;i < 4;i++) {
+        char addr[4];
+        char* ptr = addr;
+        do {
```

Comment:
That's one hell of a confusing way to copy a string up until a delimiter :thinking: